arch/arm/soc/atmel_sam0/samd/soc.c

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (c) 2017 Google LLC.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Atmel SAMD MCU series initialization code
+ */
+
+#include <arch/cpu.h>
+#include <cortex_m/exc.h>
+#include <device.h>
+#include <init.h>
+#include <kernel.h>
+#include <soc.h>
+
+void soc_gpio_configure(const struct soc_gpio_pin *pin)
```

Comment:
why aren't we using the gpio interface instead?

Reply:
We would need to use the pinmux driver, which - as it is - is not very handy.

One thing that pinmux driver is currently unable to do is to program a few pins which belong to the same port in one go. Very handy when setting up e.g. Ethernet driver or any other module with multiple pins. Atmel typically puts all pins belonging to a certain peripheral on the same mux of the same port. Having a dedicated private function makes setting up pins very easy.

We should certainly provide pinmux driver for the user in the future. However, currently pinmux API is not documented. In fact the whole GPIO/pinmux driver combo seems to be a bit of a mess.

Reply:
+1 to what @mnkp said.  I'm happy to add pinmux support in a follow up CL.

Reply:
Err, 'pull request', not CL.  Wrong source code control system :)

Reply:
I'm ok with that plan.